R/appMetadata.R

```diff
@@ -6,32 +6,27 @@ appMetadata <- function(appDir,
                         isCloudServer = FALSE,
                         metadata = list()) {
 
-  if (is.null(appFiles)) {
-    appFiles <- bundleFiles(appDir)
-  }
+  appFiles <- standardizeAppFiles(appDir, appFiles)
 
-  if (!is.null(contentCategory)) {
-    assetTypeName <- contentCategory
-  } else if (!is.null(appPrimaryDoc)) {
-    assetTypeName <- "document"
+  # User has supplied quarto path or quarto package has supplied metadata
+  # https://github.com/quarto-dev/quarto-r/blob/08caf0f42504e7/R/publish.R#L117-L121
+  hasQuarto <- !is.null(quarto) || hasQuartoMetadata(metadata)
+
+  # Generally we want to infer appPrimaryDoc from appMode, but there's one
+  # special case
+  if (!is.null(appPrimaryDoc) &&
+      tolower(tools::file_ext(appPrimaryDoc)) == "r") {
+    appMode <- "shiny"
   } else {
-    assetTypeName <- "application"
+    # Inference only uses top-level files
+    rootFiles <- appFiles[dirname(appFiles) == "."]
+    appMode <- inferAppMode(
+      file.path(appDir, appFiles),
```

Comment:
Does `inferAppMode` require absolute paths? Should we rename the argument to make it easier for the reader to remember that this instance has `absoluteAppFiles` rather than the relative paths present in `appFiles`?

Reply:
That variable was my attempt at conveying that impression 😄 but I like yours better.